src/CacheManager.Microsoft.Extensions.Caching.Memory/MemoryCacheHandle`1.cs

```diff
@@ -152,10 +173,7 @@ protected override bool AddInternalPrepared(CacheItem<TCacheValue> item)
             var options = GetOptions(item);
             _cache.Set(key, item, options);
 
-            if (item.Region != null)
-            {
-                _cache.RegisterChild(item.Region, key);
-            }
+            _cache.RegisterChild(item.Region ?? GetType().Name, key);
```

Comment:
Adding and Removing keys was handled in the original commit.

Given it's non-trivial, it might make sense to put it on a configuration option.